php/libraries/DataFrameworkMenu.class.inc

```diff
@@ -183,5 +183,91 @@ abstract class DataFrameworkMenu extends NDB_Menu_Filter
         return new \LORIS\Http\DataIteratorBinaryStream($rows);
 
     }
+
+    /**
+     * Format the list of user's projects to display the module filter options
+     *
+     * @param bool $numericKeys If the returned projects keys
+     *                          should be numeric (ProjectID) or project name
+     *
+     * @return array - an associative array of the form (ID|mame => name)
+     */
+    public function getProjectOptions(
+        bool  $numericKeys = true
+    ): array {
+        $projects     = \NDB_Factory::singleton()->user()->getProjects();
+        $projectNames = array_map(
+            fn($project) => $project->getName(),
+            $projects
+        );
+        if ($numericKeys) {
+            $projectIDs = array_map(
+                fn($project) => strval($project->getId()),
+                $projects
+            );
+            return array_combine($projectIDs, $projectNames);
+        } else {
+            return array_combine($projectNames, $projectNames);
+        }
+    }
+
+    /**
+     * Get the list of subprojects for which the current user has permission
+     * To populate the module subprojects filter options
+     *
+     * @param bool $numericKeys If the returned subprojects keys
+     *                          should be numeric (SubprojectID)
+     *                          or strings (subproject name)
+     *
+     * @return array - an associative array of the form (ID => name)
+     */
+    public function getSubprojectOptions(
+        bool $numericKeys = true
+    ): array {
+        $subprojectOptions = array_reduce(
+            \NDB_Factory::singleton()->user()->getProjects(),
+            fn($subprojects, $project) =>
+                $subprojects
+                + array_column($project->getSubprojects(), 'title', 'subprojectId'),
+            []
+        );
+
+        if (!$numericKeys) {
+            $subprojectNames = array_values($subprojectOptions);
+            return array_combine($subprojectNames, $subprojectNames);
+        }
```

Comment:
in the above function you have
```
if (..) {
   return 1
} else {
   return 2
}
```

But in this one the style is:

```
if (...) {
    return 1
}
return 2
```

You should be consistent (Stylistically I prefer the second, but as long as it's consistent it doesn't really matter which..)